src/main/java/bio/overture/songsearch/graphql/GraphQLProvider.java

```diff
@@ -48,7 +60,18 @@ public void init() throws IOException {
     URL url = Resources.getResource("schema.graphql");
     String sdl = Resources.toString(url, Charsets.UTF_8);
     graphQLSchema = buildSchema(sdl);
-    this.graphQL = GraphQL.newGraphQL(graphQLSchema).build();
+    this.graphQL = buildGraphql(graphQLSchema);
+  }
+
+  private GraphQL buildGraphql(GraphQLSchema graphQLSchema) {
+      val graphQLBuilder = GraphQL.newGraphQL(graphQLSchema);
+      if (authProperties.isEnabled()) {
+          // For more info on `Execution Strategies` see: https://www.graphql-java.com/documentation/v15/execution/
```

Comment:
Java annotations don't function the same as python/typescript decorators, so I just did traditional class decorators. Also I found a better place to apply this logic, here in the actual query execution strategy so we don't have to put it on each individual data fetcher.

Reply:
After further discussions, we agree that this all or nothing approach is good with our current spec. and can be refactored if the approach is changed in the future.